### PR TITLE
boards: arm: nucleo_l152re: add support for spi

### PR DESCRIPTION
**Pull request by HoZHel**
Defined SPI1, SPI2, and SPI3 pin controls for Nucleo-L152RE.
Defined arduino_spi.
Add definition of SPI3 for STM32L1 series.

### CODE REVIEW

File: dts/arm/st/l1/stm32l1.dtsi [hunk 1]
```diff
@@ -200,6 +200,16 @@
 			status = "disabled";
 		};
 
+    spi3: spi@40003c00 {
```
Comment (erwango) [inline dts/arm/st/l1/stm32l1.dtsi:203]: Please fix indentation

**Review by erwango — CHANGES_REQUESTED** [review]

File: dts/arm/st/l1/stm32l1.dtsi (hunk 1 above)
Comment (gautierg-st) [inline dts/arm/st/l1/stm32l1.dtsi:203]: SPI3 is not available on the complete STM32L1x series but only on STM32L1xxxC/D/E.
So this definition should be added in stm32l151Xc.dtsi, stm32l152Xc.dtsi and stm32l152Xe.dtsi (like it is done for timers5 for example)

**Review by gautierg-st — CHANGES_REQUESTED** [review]

File: dts/arm/st/l1/stm32l1.dtsi (hunk 1 above)
Comment (HoZHel) [reply dts/arm/st/l1/stm32l1.dtsi]: Done. Thanks, see also https://github.com/zephyrproject-rtos/zephyr/pull/61656#discussion_r1299861203

Comment (HoZHel) [reply dts/arm/st/l1/stm32l1.dtsi]: Done. Thanks for the suggestion.

**Review by gautierg-st — APPROVED** [review]
Thanks

**Review by erwango — APPROVED** [review]

**Pull request merged**